zookeeper-server/src/main/java/org/apache/zookeeper/server/ServerCnxn.java

```diff
@@ -82,6 +80,22 @@ public ServerCnxn(final ZooKeeperServer zkServer) {
         this.zkServer = zkServer;
     }
 
+    /**
+     * Flag that indicates that this connection is known to be closed/closing
+     * and from which we can optionally ignore outstanding requests as part
+     * of request throttling. This flag may be false when a connection is
+     * actually closed (false negative), but should never be true with
+     * a connection is still alive (false positive).
```

Comment:
This makes sense, though I've seen weird states of heap dumps where connections were set as stale but not closed (aka stale but alive - false positive). Unfortunately I was unable to reproduce this or identify the code culprit. All I am saying is to be cautious on relying the stale flag inside the server connection object...

Reply:
Thanks for sharing your finding. It seems working for us so far. Will keep an eye one it.